test/dotcom/system_status/alerts_test.exs

```diff
@@ -0,0 +1,178 @@
+defmodule Dotcom.SystemStatus.AlertsTest do
+  use ExUnit.Case, async: true
+  doctest Dotcom.SystemStatus.Alerts
+
+  import Test.Support.Factories.Alerts.Alert
+
+  alias Dotcom.SystemStatus.Alerts
+
+  defp local_datetime(naive) do
+    DateTime.from_naive!(naive, "America/New_York")
+  end
+
+  describe "active_today?/2" do
+    test "returns true if the alert is currently active" do
+      assert Alerts.active_today?(
+               build(:alert,
+                 active_period: [
+                   {
+                     local_datetime(~N[2025-01-09 12:00:00]),
+                     local_datetime(~N[2025-01-09 20:00:00])
+                   }
+                 ]
+               ),
+               local_datetime(~N[2025-01-09 13:00:00])
+             )
+    end
+
+    test "returns false if the alert starts after end-of-service" do
+      refute Alerts.active_today?(
+               build(:alert,
+                 active_period: [
+                   {
+                     local_datetime(~N[2025-01-10 12:00:00]),
+                     local_datetime(~N[2025-01-10 20:00:00])
+                   }
+                 ]
+               ),
+               local_datetime(~N[2025-01-09 13:00:00])
+             )
+    end
+
+    test "returns true if the alert starts later, but before end-of-service" do
+      assert Alerts.active_today?(
+               build(:alert,
+                 active_period: [
+                   {
+                     local_datetime(~N[2025-01-09 20:00:00]),
+                     local_datetime(~N[2025-01-10 20:00:00])
+                   }
+                 ]
+               ),
+               local_datetime(~N[2025-01-09 13:00:00])
+             )
+    end
+
+    test "returns false if the alert has already ended" do
+      refute Alerts.active_today?(
+               build(:alert,
+                 active_period: [
+                   {
+                     local_datetime(~N[2025-01-09 10:00:00]),
+                     local_datetime(~N[2025-01-09 12:00:00])
+                   }
+                 ]
+               ),
+               local_datetime(~N[2025-01-09 13:00:00])
+             )
+    end
+
+    test "returns true if the alert has no end time" do
+      alert =
+        build(:alert,
```

Comment:
This file might read a bit cleaner with a helper function to generate an alert from a given active_period start and/or stop 😅

Reply:
@thecristen I rewrote a bunch of the tests per Anthony's feedback above. Do you still feel this way, or do you think that pulling out a bunch of `start_time`/`end_time` variables may have helped?

Reply:
It totally helps!